gitmod/matomo.js

```diff
@@ -1,33 +1,72 @@
+    if (window.gl !== 'undefined' && document.readyState == "complete" && document.querySelectorAll('[data-project]').length) {
+      clearInterval(checkRequiredElementsExist);
+      hideThings();
+      gainsightIdentify();
+    }
+  }, 100);
+
+var oldURL = window.location.href;
+setInterval(checkURLchange, 1000);
+
+/**
+ * Check if there is change in URL. If the change observed invoke hideThings();
+ *
+ */ 
+function checkURLchange(){
+  if(window.location.href != oldURL){
+      oldURL = window.location.href;
+      hideThings();
   }
-}, 100);
+}
 
+/**
+ * Add logic to hide the webide and edit options from Code Studio UI
+ *
+ */ 
 function hideThings () {
-  var webIdeButton = document.querySelector('[data-qa-selector="action_dropdown"]')
-  if(webIdeButton){
-    webIdeButton.setAttribute('style', 'display:none !important')
-  }
-  var editWebButton = document.querySelector('[data-qa-selector="webide_menu_item"]')
-  if (editWebButton) {
-    editWebButton.setAttribute('style', 'display:none !important')
+  // Fetch the document that contains 'Web IDE' text
+  var webIde = document.evaluate("//span[contains(., 'Web IDE')]", document, null, XPathResult.ANY_TYPE, null );
+  var webIdeDoc = webIde.iterateNext();
+  // Assign the document to webIdeDiv
+  var webIdeDiv = webIdeDoc;
+  var content;
+  var hasText;
+  // Iterate to get the parent node of the webIdeDiv, length is set to 10 as the currently
+  // it takes 9 iterations to get the parent div where we want to apply style
+  for (let index = 1; index < 10; index++) {
+    content = webIdeDiv.textContent || webIdeDiv.innerText;
+    // extra check
+    hasText = content.toLowerCase().includes("edit");
+    // Add checks to verify if the correct parent div is taken
+    if (((webIdeDiv.classList.contains("gl-new-dropdown")) && (webIdeDiv.classList.contains("gl-display-block!")) && hasText)) {
+      // As there are 2 places for webide, and dropdown length differs for both the places
+      // for multipe list, style is applied on list
+      if(webIdeDiv.querySelectorAll("li").length > 2){
+        var closeseList=webIdeDoc.closest("li");
+        closeseList.setAttribute('style', 'display:none !important');
+      }
+      webIdeDiv.setAttribute('style', 'display:none !important');
+    } else {
+      webIdeDiv = webIdeDiv.parentNode
+    }
   }
+  // Hide Operator section from left panel
   if ((operateLink = document.querySelector('[data-qa-section-name="Operate"]'))) {
     operateLink.setAttribute('style', 'display:none !important')
   }
+  // Hide Monitor section from left panel
   if ((monitorLink = document.querySelector('[data-qa-section-name="Monitor"]'))) {
     monitorLink.setAttribute('style', 'display:none !important')
   }
+  // Hide 'Add Kubernetes cluster' section from project page
   if ((k8sLink = document.evaluate(
-        "//a[contains(text(),'Add Kubernetes cluster')]", document, null,
+        "//a[contains(.,'Add Kubernetes cluster')]", document, null,
```

Comment:
@bansodejoyce just making sure this change is intended and works as expected stil?

Reply:
This is not working currently